modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/CatchNode.java

```diff
@@ -96,15 +92,14 @@ protected void write(ClassWriter classWriter, MethodWriter methodWriter, WriteSc
         methodWriter.visitVarInsn(variable.getAsmType().getOpcode(Opcodes.ISTORE), variable.getSlot());
 
         if (blockNode != null) {
-            blockNode.continueLabel = continueLabel;
-            blockNode.breakLabel = breakLabel;
-            blockNode.write(classWriter, methodWriter, writeScope);
+            blockNode.write(writeScope.newTryScope(null, null, null));
```

Comment:
This is weird.

Reply:
Agreed, updated scoping to improve the handling for this unique case.